File: algorithm_practice_typescript/1720.ts

function decode(encoded: number[], first: number): number[] {
    const ans = [first];
    for (const i of encoded) {
        const last = ans[ans.length - 1];
        ans.push(last ^ i);
    }
    return ans;
};

function sol2(encoded: number[], first: number): number[] {
    const ans = [first];
    for (let i = 0; i < encoded.length; i++) {
        ans.push(ans[i] ^ encoded[i])
    }
    return ans;
};

const encoded = [1,2,3]
const first = 1

console.log(sol2(encoded, first))

File: algorithm_practice_python/leetcode/arrays101/1720.py
from typing import List


class Solution:
    def decode(self, encoded: List[int], first: int) -> List[int]:
        arr = [first]
        for idx, i in enumerate(encoded):
            arr.append(i ^ arr[idx])
        return arr

    def sol2(self, encoded, first):
        arr = [first]
        for i in encoded:
            arr.append(arr[-1] ^ i)
        return arr


if __name__ == '__main__':
    encoded = [1, 2, 3]
    first = 1

    ans = Solution().sol2(encoded, first)
    print(f'{ans}, [1,0,2,1]')
